tests/common/reboot.py

```diff
@@ -120,14 +128,22 @@ def execute_reboot_helper():
         logger.info('rebooting {} with helper "{}"'.format(hostname, reboot_helper))
         return reboot_helper(reboot_kwargs)
 
+    def trigger_kernel_crash_and_reboot():
+        logger.info("Triggering kernel panic with command '{}' ..."
+                    .format(reboot_command))
+        return duthost.shell(reboot_command)
+
     dut_datetime = duthost.get_now_time()
     DUT_ACTIVE.clear()
 
-    if reboot_type != REBOOT_TYPE_POWEROFF:
-        reboot_res = pool.apply_async(execute_reboot_command)
-    else:
+    if reboot_type == REBOOT_TYPE_KDUMP:
+        reboot_res = pool.apply_async(trigger_kernel_crash_and_reboot)
```

Comment:
Why do we need a special case here? Why not simply add the logger.info() line to the `if reboot_type != REBOOT_TYPE_POWEROFF:` case, so that we log every type of reboot?

Reply:
Good question! Initially I want to use the function `execute_reboot_command()` to trigger the kernel crash and reboot. But I found the `duthost.command(...)` was unable to execute the command `echo c > sysrq-trigger` successfully. 

So I used the `duthost.shell(...)` to execute the command. Also I did not use the `duthost.shell(...)` to replace the `duthost.command(...)` in the function `execute_reboot_command()` because I am not sure whether such change will introduce the regression or not.

Reply:
Did you try `echo 'c' > sysrq-trigger` (put the `c` in quotes) with `duthost.command(...)`? I don't know what else could make a difference.

Reply:
Let me try this ...